src/page/bell/Button.ts

```diff
@@ -2,154 +2,170 @@ import { addDomElement, removeDomElement } from 'src/shared/helpers/dom';
 import Bell from './Bell';
-import Message from './Message';
+import { MESSAGE_TIMEOUT, MessageType } from './constants';
 
-export default class Button extends ActiveAnimatedElement {
-  public events: any;
-  public bell: Bell;
+export default class Button extends AnimatedElement {
+  public _events: Record<string, string>;
+  public _bell: Bell;
+  private _isHandlingClick = false;
 
   constructor(bell: Bell) {
     super(
       '.onesignal-bell-launcher-button',
       undefined,
-      undefined,
       'onesignal-bell-launcher-button-active',
       undefined,
-      'shown',
-      '',
     );
 
-    this.bell = bell;
-    this.events = {
+    this._bell = bell;
+    this._events = {
       mouse: 'bell.launcher.button.mouse',
     };
 
-    const element = this.element;
+    const element = this._element;
     if (element) {
       element.addEventListener(
         'touchstart',
         () => {
-          this.onHovering();
-          this.onTap();
+          this._onHovering();
+          this._onTap();
         },
         { passive: true },
       );
 
       element.addEventListener('mouseenter', () => {
-        this.onHovering();
+        this._onHovering();
       });
 
       element.addEventListener('mouseleave', () => {
-        this.onHovered();
+        this._onHovered();
       });
       element.addEventListener(
         'touchmove',
         () => {
-          this.onHovered();
+          this._onHovered();
         },
         { passive: true },
       );
 
       element.addEventListener('mousedown', () => {
-        this.onTap();
+        this._onTap();
       });
 
       element.addEventListener('mouseup', () => {
-        this.onEndTap();
+        this._onEndTap();
       });
 
       element.addEventListener('click', () => {
-        this.onHovered();
-        this.onClick();
+        this._onClick();
       });
     }
   }
 
-  onHovering() {
+  _onHovering() {
     if (
-      LimitStore.isEmpty(this.events.mouse) ||
-      LimitStore.getLast(this.events.mouse) === 'out'
+      LimitStore.isEmpty(this._events.mouse) ||
+      LimitStore.getLast(this._events.mouse) === 'out'
     ) {
-      OneSignalEvent.trigger(Bell.EVENTS.HOVERING);
+      OneSignalEvent.trigger('notifyButtonHovering');
     }
-    LimitStore.put(this.events.mouse, 'over');
+    LimitStore.put(this._events.mouse, 'over');
   }
 
-  onHovered() {
-    LimitStore.put(this.events.mouse, 'out');
-    OneSignalEvent.trigger(Bell.EVENTS.HOVERED);
+  _onHovered() {
+    LimitStore.put(this._events.mouse, 'out');
+    OneSignalEvent.trigger('notifyButtonHover');
   }
 
-  onTap() {
-    this.pulse();
-    this.activate();
-    this.bell.badge.activate();
+  _onTap() {
+    this._pulse();
+    this._activate();
+    this._bell._badge._activate();
   }
 
-  onEndTap() {
-    this.inactivate();
-    this.bell.badge.inactivate();
+  _onEndTap() {
+    this._inactivate();
+    this._bell._badge._inactivate();
```

Comment:
Including changes like these in the same commit make it difficult to parse out the parts that actually affect Bell behavior. I generally recommend keeping any refactoring changes to their own separate commits.